frontend/catalyst/pennylane_extensions.py

```diff
@@ -247,8 +206,12 @@ class QJITDevice(qml.QubitDevice):
         "Hamiltonian",
     ]
 
-    def __init__(self, shots=None, wires=None, backend_name=None, backend_kwargs=None):
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self, shots=None, wires=None, backend_name=None, backend_lib=None, backend_kwargs=None
+    ):
         self.backend_name = backend_name if backend_name else "default"
+        self.backend_lib = backend_lib if backend_lib else ""
```

Comment:
Are we using this new property anywhere?

Reply:
not precisely! As the `backend_lib` variable must always initialize with or without an absolute path to the shared library of the backend device.

Reply:
Where is the property used?

Reply:
This property contains the name of the shared library (that's different from the name of the device which stores in `backend_name`). All these backend info will be traced and hardcoded to the generated JAXPR by the `qdevice` primitive. The runtime uses the value of `backend_lib` to link to the library and `backend_name` to resolve the symbol of the factory method.

Reply:
Ah got it, thanks!